### ISSUE TITLE
Retinanet get loss_cls: 31852.0547, loss_bbox: nan, loss: nan

### ISSUE THREAD

**Issue by Hukongtao** [open-mmlab/mmdetection#4229]
**Describe the issue**

After install mmdet, I want to train a retinanet model using coco dataset. After hundreds of iterations，The network gradually starts to not converge

**Reproduction**
1. What command or script did you run?
```
python tools/train.py configs/retinanet/retinanet_r50_fpn_1x_coco.py
```
2. What config dir you run?
```
configs/retinanet/retinanet_r50_fpn_1x_coco.py
```
3. Did you make any modifications on the code or config? Did you understand what you have modified?
```
I changed nothing, except the path of dataset
```
4. What dataset did you use?
```
coco dataset, train2017 and val2017
```

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.
```
sys.platform: linux
Python: 3.6.6 (default, Jan 26 2019, 16:53:05) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
CUDA available: True
GPU 0,1,2,3: TITAN V
CUDA_HOME: /usr/local/cuda-9.2
NVCC: Cuda compilation tools, release 9.2, V9.2.148
GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
PyTorch: 1.5.0+cu92
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2019.0.5 Product Build 20190808 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 9.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.6.0+cu92
OpenCV: 4.4.0
MMCV: 1.2.0
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 9.2
MMDetection: 2.6.0+
```

2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]
    ```
    pip
    ```
    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Results**

If applicable, paste the related results here, e.g., what you expect and what you get.
```
2020-12-04 10:42:46,201 - mmdet - INFO - workflow: [('train', 1)], max: 12 epochs
2020-12-04 10:43:01,199 - mmdet - INFO - Epoch [1][50/58633]	lr: 9.890e-04, eta: 2 days, 10:23:38, time: 0.299, data_time: 0.061, memory: 3325, loss_cls: 1.2171, loss_bbox: 0.6967, loss: 1.9139
2020-12-04 10:43:13,421 - mmdet - INFO - Epoch [1][100/58633]	lr: 1.988e-03, eta: 2 days, 5:04:36, time: 0.244, data_time: 0.009, memory: 3325, loss_cls: 1.2216, loss_bbox: 0.6977, loss: 1.9193
2020-12-04 10:43:25,787 - mmdet - INFO - Epoch [1][150/58633]	lr: 2.987e-03, eta: 2 days, 3:28:54, time: 0.247, data_time: 0.009, memory: 3326, loss_cls: 1.2253, loss_bbox: 0.6915, loss: 1.9169
2020-12-04 10:43:37,790 - mmdet - INFO - Epoch [1][200/58633]	lr: 3.986e-03, eta: 2 days, 2:20:21, time: 0.240, data_time: 0.008, memory: 3326, loss_cls: 1.2254, loss_bbox: 0.6790, loss: 1.9044
2020-12-04 10:43:49,785 - mmdet - INFO - Epoch [1][250/58633]	lr: 4.985e-03, eta: 2 days, 1:38:39, time: 0.240, data_time: 0.008, memory: 3326, loss_cls: 1.1934, loss_bbox: 0.6909, loss: 1.8842
2020-12-04 10:44:01,915 - mmdet - INFO - Epoch [1][300/58633]	lr: 5.984e-03, eta: 2 days, 1:15:58, time: 0.243, data_time: 0.009, memory: 3326, loss_cls: 1.1883, loss_bbox: 0.7420, loss: 1.9302
2020-12-04 10:44:13,893 - mmdet - INFO - Epoch [1][350/58633]	lr: 6.983e-03, eta: 2 days, 0:54:32, time: 0.240, data_time: 0.008, memory: 3326, loss_cls: 1.1919, loss_bbox: 0.6690, loss: 1.8609
2020-12-04 10:44:26,157 - mmdet - INFO - Epoch [1][400/58633]	lr: 7.982e-03, eta: 2 days, 0:46:59, time: 0.245, data_time: 0.008, memory: 3326, loss_cls: 1.1792, loss_bbox: 0.6646, loss: 1.8438
2020-12-04 10:44:38,885 - mmdet - INFO - Epoch [1][450/58633]	lr: 8.981e-03, eta: 2 days, 0:52:53, time: 0.254, data_time: 0.008, memory: 3326, loss_cls: 1.2042, loss_bbox: 0.6764, loss: 1.8806
2020-12-04 10:44:51,571 - mmdet - INFO - Epoch [1][500/58633]	lr: 9.980e-03, eta: 2 days, 0:56:43, time: 0.254, data_time: 0.007, memory: 3326, loss_cls: 1.1566, loss_bbox: 0.6873, loss: 1.8439
2020-12-04 10:45:04,080 - mmdet - INFO - Epoch [1][550/58633]	lr: 1.000e-02, eta: 2 days, 0:56:02, time: 0.250, data_time: 0.007, memory: 3326, loss_cls: 1.1940, loss_bbox: 0.7149, loss: 1.9089
2020-12-04 10:45:16,692 - mmdet - INFO - Epoch [1][600/58633]	lr: 1.000e-02, eta: 2 days, 0:57:19, time: 0.252, data_time: 0.008, memory: 3326, loss_cls: 31852.0547, loss_bbox: nan, loss: nan
```

**Issue fix**

If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by ZwwWayne** [comment]
It seems that you only use a single GPU to train the model with actual batch size 2. However, the learning rate is 0.01 and is used for 8 GPU training with a batch size of 16, which is too large for your case. We suggest you use a smaller learning rate, e.g., 0.01/8. This should be the main issue, if you still meet the problem after adopting a small LR, we suggest you follow the [doc](https://mmdetection.readthedocs.io/en/latest/faq.html#training).

**Comment by tianxianhao** [comment]
I also meet the same problem. My batch size is 16, but loss is also nan.

**Comment by tianxianhao** [comment]
The problem occurs when i update the mmcv version from 1.2.4 to 1.2.7.

**Comment by chienhust** [comment]
I also meet the same problem. Can you show me how to fix it?

**Comment by countytown** [comment]
The problem still exists at mmdetection2.25.2, so disappointed. And any detector that inherits RetinaNet will has the same problem.
